pxtlib/emitter/emitter.ts

```diff
@@ -488,6 +488,15 @@ namespace ts.pxtc {
             return cached
         let res = parseCommentString(getComments(node))
         res._name = getName(node)
+        if (node.kind == SK.FunctionDeclaration && res.block === "true" && !res.blockId) {
+            const fn = (node as any) as ts.FunctionDeclaration;
+            if ((fn.symbol as any).parent) {
+                res.blockId = `${(fn.symbol as any).parent.name}_${getDeclName(fn)}`;
+                res.block = `${node.symbol.name}${fn.parameters.length ? '|' + fn.parameters
+                    .filter(p => !p.questionToken)
+                    .map(p => `${p.symbol.name} %${p.symbol.name}`).join('|') : ''}`;
```

Comment:
Instead of `p.symbol.name` you should use `(p.name as ts.Identifier).text` as `symbol` might not be defined. Also, error out here on a binding pattern.